### PR TITLE
altering the sort to avoid an exception error and adding is array check

### PR DESCRIPTION
**Pull request by leonoragilmour**
I have had to make 2 changes to accommodate the post type picker

- I have changed the sort method. The usort would throw an exception. Now it sorts the array with the keys which I set when the widget is added to the array.

- I have to check it is an array before checking what's in the array.

- Plugin PR https://github.com/shortlist-digital/wp-widget-post-type-picker-plugin/pull/1

- Croissant Heavy PR https://github.com/shortlist-digital/croissant-heavy/pull/48

### CODE REVIEW

File: custom-fields/WidgetLoader.php [hunk 1]
```diff
@@ -89,8 +80,10 @@ protected static function traverseHierarchy($path) {
       $fullpath = $path . '/' . $file;
       if(is_dir($fullpath))
         $return_array = array_merge($return_array, self::traverseHierarchy($fullpath));
-      elseif(substr($file, -7) == "acf.php")
-        $return_array[] = $fullpath;
+      elseif(substr($file, -7) == "acf.php") {
+        include $fullpath;
```
Comment (ecoad) [inline custom-fields/WidgetLoader.php]: I'm not sure about including the file at this point. This traverse function should just build up an array of paths. Same issue with the other `traversePlugins`

Comment (leonoragilmour) [inline custom-fields/WidgetLoader.php]: I know what you mean - is there a better way to sort the array by it's label?
Previously, the file was being included within the sort, and I think that's where the exception was thrown, ...it was perhaps circular...

File: custom-fields/WidgetLoader.php [hunk 2]
```diff
@@ -89,8 +80,12 @@ protected static function traverseHierarchy($path) {
...
       if(is_dir($fullpath))
         $return_array = array_merge($return_array, self::traverseHierarchy($fullpath));
-      elseif(substr($file, -7) == "acf.php")
-        $return_array[] = $fullpath;
+      elseif(substr($file, -7) == "acf.php") {
+        $path_array = explode("/", $fullpath);
+        end($path_array);
+        $label = prev($path_array);
+        $return_array[$label] = $fullpath;
+      }
```
Comment (leonoragilmour) [inline custom-fields/WidgetLoader.php]: @ecoad what do you think of this instead?

File: custom-fields/WidgetLoader.php [hunk 3]
```diff
@@ -106,7 +101,9 @@ protected static function traversePlugins() {
       $fullpath = plugin_dir_path(WP_CONTENT_DIR.'/plugins/'.$file).'widget-loader-acf.php';
       // If plugin directory contains acf.php file and is agreable prefixed...
       if(substr($file, 0, 8) === "agreable" && file_exists($fullpath)){
-        $return_array[] = $fullpath;
+        $path_array = explode("/", $file);
+        $label = substr($path_array[0], 9);
+        $return_array[$label] = $fullpath;
```
Comment (leonoragilmour) [inline custom-fields/WidgetLoader.php]: @ecoad and a bit different for this one

**Pull request merged**